sparse/core.py

```diff
@@ -701,12 +719,16 @@ def elemwise_binary(self, func, other, *args, **kwargs):
         coords = np.concatenate(coords_list, axis=1) if len(coords_list) else \
             np.empty((0, len(result_shape)), dtype=self.coords.dtype)
 
-        nonzero = data != 0
+        nonzero = data != func_zero
         data = data[nonzero]
         coords = coords[:, nonzero]
 
         return COO(coords, data, shape=result_shape, has_duplicates=False)
 
+    @staticmethod
+    def _zero_of_dtype(dtype):
+        return np.zeros((), dtype=dtype)
```

Comment:
I have a slight preference to move utility functions like this to be normal functions rather than staticmethods.

Reply:
Would you prefer it be in `utils.py` or `core.py`?